src/components/App/ExpressRecorder.tsx

```diff
@@ -447,9 +447,9 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
         // check if something has been changed
         const { constraints, shareScreenOn } = this.state;
         if (
-            ((selectedCamera && constraints.video) || // check if video was turn on/off
+            ((selectedCamera && constraints.video) || // check if video has turned on/off
                 (!selectedCamera && !constraints.video)) &&
-            ((selectedAudio && constraints.audio) || // check if audio was turn on/off
+            ((selectedAudio && constraints.audio) || // check if audio has turned on/off
                 (!selectedAudio && !constraints.audio)) &&
             (!selectedCamera || // check if device id has been changed
                 (constraints.video &&
```

Comment:
it's easy to understand why you got confused with this condition.
it is horrible typing and should be refactored asap.